cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```
@@ -440,13 +443,56 @@ public void getArtifactPlugins(HttpRequest request, HttpResponder responder,
+    }
+  }
+
+  @Beta
+  @POST
+  @Path("/namespaces/{namespace-id}/artifacts/{artifact-name}/" +
+    "versions/{artifact-version}/plugintypes/{plugin-type}/plugins/{plugin-name}/methods/{plugin-method}")
+  public void callArtifactPluginMethod(HttpRequest request, HttpResponder responder,
+                                       @PathParam("namespace-id") String namespaceId,
+                                       @PathParam("artifact-name") String artifactName,
+                                       @PathParam("artifact-version") String artifactVersion,
+                                       @PathParam("plugin-name") String pluginName,
+                                       @PathParam("plugin-type") String pluginType,
+                                       @PathParam("plugin-method") String methodName,
+                                       @QueryParam("scope") @DefaultValue("user") String scope)
+    throws NamespaceNotFoundException, BadRequestException, ArtifactNotFoundException, IOException {
+
+    String requestBody = request.getContent().toString(Charsets.UTF_8);
+    Id.Namespace namespace = Id.Namespace.from(namespaceId);
+    Id.Namespace artifactNamespace = validateAndGetScopedNamespace(namespace, scope);
+    Id.Artifact artifactId = validateAndGetArtifactId(artifactNamespace, artifactName, artifactVersion);
+    if (requestBody.isEmpty()) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Request body is used as plugin method parameter, " +
+        "Received empty request body. ");
+      return;
+    }
+    try {
+      String response =
+        artifactRepository.pickParentArtifactAndInvokePluginEndpoint(namespace, artifactId, pluginType,
+                                                                     pluginName, methodName, requestBody);
+      responder.sendJson(HttpResponseStatus.OK, response);
+    } catch (NotFoundException | IllegalArgumentException e) {
+      LOG.error("Exception when invoking plugin method", e);
```

why does an illegal argument exception translate to a not found?  Also, you don't have to catch this, you can just let this bubble up and the framework will take care of returning a 404.